gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java

```diff
@@ -668,16 +670,26 @@ public FSDataOutputStream create(
 
     OutputStreamType type = GCS_OUTPUT_STREAM_TYPE.get(getConf(), getConf()::getEnum);
     OutputStream out;
+    int minSyncTimeIntervalMs;
     switch (type) {
       case BASIC:
         out =
             new GoogleHadoopOutputStream(
                 this, gcsPath, statistics, new CreateFileOptions(overwrite));
         break;
+      case FLUSHABLE_COMPOSITE:
+        minSyncTimeIntervalMs = GCS_OUTPUT_STREAM_SYNC_MIN_INTERVAL_MS.get(
+            getConf(), getConf()::getInt);
+        out =
+            new GoogleHadoopSyncableOutputStream(
+                this, gcsPath, statistics, new CreateFileOptions(overwrite), false, true, minSyncTimeIntervalMs);
```

Comment:
We need to create an option class for `GoogleHadoopSyncableOutputStream` similar to `AsyncWriteChannelOptions` with `minSyncInterval`, `appendEnabled` and `syncOnFlushEnabled` properties to make code more readable.

Reply:
done.